src/cargo/core/profiles.rs

```diff
@@ -539,10 +539,16 @@ pub enum ProfileRoot {
 
 /// Profile settings used to determine which compiler flags to use for a
 /// target.
-#[derive(Clone, Eq, PartialOrd, Ord, serde::Serialize)]
+#[derive(Clone, Eq, PartialOrd, Ord, serde::Serialize, derivative::Derivative)]
+/// Don't compare/hash fields which wont affect compilation.
+/// This is necessary for `Unit` deduplication for things like "test" and
+/// "dev" which are essentially the same.
+#[derivative(Hash, PartialEq)]
 pub struct Profile {
+    #[derivative(Hash = "ignore", PartialEq = "ignore")]
```

Comment:
Could you re-do the commits so that one commit is the refactor with the other commit having the fix?  And describe the problem and the fix in the PR?

This is putting the burden on the reviewer to figure out what is happening with this and the team is already strapped for availability.